API Blueprint Specification.md

```diff
@@ -478,6 +490,232 @@ Name and description of the API
+        + <customizable option N>
+
+Where:
+
++ `<scheme name>` is the authentication scheme name by which this scheme is represented in [Authenticated section](#def-authenticated-section).
++ `<description>` is any **optional** Markdown-formatted description of the authentication scheme.
++ `<base scheme type>` is the base authentication scheme type (e.g. "Basic Authentication Scheme"). See below for all the available base authentication schemes.
++ `<customizable option n>` represents a customizable option in the base authentication scheme. It is a [MSON Property Member Declaration][] without the [MSON Type Definition][] where the [MSON Property Name][] represents the name of the option and [MSON Value][] represents the value of the option.
+
+List of the base authentication schemes supported by API Blueprint are described below.
+
+#### Basic Authentication Scheme
+This is the [Basic Authentication Scheme](http://tools.ietf.org/html/rfc2617#section-2) as specified in [RFC 2617](http://tools.ietf.org/html/rfc2617).
+
+The customizable options for this scheme are:
+
++ `example_username` represents an example username value. Value type is **string**. "example_user" is the **default**.
++ `example_password` represents an example password value. Value type is **string**. "example_pass" is the **default**.
+
+##### Example
+
+    + basic (Basic Authentication Scheme)
+        + example_username: `john`
+        + example_password: `password`
+
+#### OAuth2 Authentication Scheme
+This scheme uses the [OAuth2 Authorization Framework](http://tools.ietf.org/html/rfc6749) specified in [RFC 6749](http://tools.ietf.org/html/rfc6749) to get an authorization grant which is then used to access the API.
+
+The customizable options for this scheme are:
+
++ `oauth_resource_endpoint` is the oauth resource endpoint where the oauth framework resides. Value type is **string**. "/" is the **default**.
++ `scopes` represents the OAuth scopes which indicate a list of permissions. Value type is **array**. By **default** no scopes are specified.
++ `authorization_endpoint` is the authorization endpoint required for the authorization grant. Value type is **string**. "/authorize" is the **default**.
++ `access_token_endpoint` is the token endpoint used by client to obtain the access token. Value type is **string**. "/access_token" is the **default**.
++ `revocation_endpoint` is the revocation endpoint used by client to revoke a token. Value type is **string**. "/revoke" is the **default**.
++ `token_header_keyword` represents the string used instead of "Bearer" in "Authorization" header field according to [RFC 6750](http://tools.ietf.org/html/rfc6750#section-2.1). Value type is **string**. "Bearer" is the **default**.
++ `token_body_keyword` represents the parameter name used instead of "access_token" in the body of an API request according to [RFC 6750](http://tools.ietf.org/html/rfc6750#section-2.2). Value type is **string**. "access_token" is the **default**.
```

Comment:
I've came across many APIs that don't support body and query authorisation tokens so I think we should have a way to state it's unsupported in a given API.

This may be unsupported due to security considerations:

> Because of the security weaknesses associated with the URI method (see Section 5), including the high likelihood that the URL containing the access token will be logged, it SHOULD NOT be used unless it is impossible to transport the access token in the "Authorization" request header field or the HTTP request entity-body. Resource servers MAY support this method.